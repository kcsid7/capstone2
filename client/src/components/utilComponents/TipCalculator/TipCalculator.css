
.TipCalculator-Info {
    margin: 0.7rem 0;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}

.TipCalculator-Info-Amount {
    margin-left: 3rem;
}


.TipCalculator-CustomTip {
    width: 3rem;
}

.TipCalculator-ButtonGroup button {
    margin: 0 0.3rem 0.7rem;
}

.TipCalculator button:hover {
    cursor: pointer;
    transform: scale(1.1);
}

.TipCalculator input {
    margin: 0 0.3rem;
    text-align: center;
}

