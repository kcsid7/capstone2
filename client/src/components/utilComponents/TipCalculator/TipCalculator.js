import React, { useState, useEffect } from "react";

import currencyFormatter from "../../../helpers/currencyFormatter";

import "./TipCalculator.css"


function TipCalculator({total, setTip}) {

    const [customTip, setCustomTip] = useState(0);

    useEffect(() => {
        if (total === 0) {
            setCustomTip(s => 0);
            setTip(s => 0);
        }
    })


    function calculateTip(tipTotal, percentage) {
        const tip = +(tipTotal * (1/100) * percentage).toFixed(2);
        setTip(s => tip);
        setCustomTip(s => tip);
    }

    function handleInputChange(e) {
        const tipNum = + e.target.value;
        setCustomTip(s => tipNum);
        setTip(s => tipNum);
    }

    return(
        <>
            <div className="TipCalculator">
                <div className="TipCalculator-Info">
                    <h6>Tip</h6>
                    <h6 className="TipCalculator-Info-Amount">{currencyFormatter(customTip)}</h6>
                </div>
                <div className="TipCalculator-ButtonGroup">
                    <button onClick={() => calculateTip(total, 12)}>12%</button>
                    <button onClick={() => calculateTip(total, 15)}>15%</button>
                    <button onClick={() => calculateTip(total, 18)}>18%</button>
                    <input className="TipCalculator-CustomTip" onChange={handleInputChange} name="custom-tip" value={customTip}></input>
                </div>
            </div>
        </>
    )
}

export default TipCalculator