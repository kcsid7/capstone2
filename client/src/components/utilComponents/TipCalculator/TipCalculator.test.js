import React from "react";
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom'; 

import TipCalculator from "./TipCalculator";


// Smoke Test
test("Renders without crashing", () => {
    render(<BrowserRouter><TipCalculator /></BrowserRouter>)
})


test("Count Button snapshot", () => {
    const { asFragment } = render(<BrowserRouter><TipCalculator /></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
})