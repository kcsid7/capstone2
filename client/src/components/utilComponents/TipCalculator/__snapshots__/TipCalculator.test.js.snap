// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count Button snapshot 1`] = `
<DocumentFragment>
  <div
    class="TipCalculator"
  >
    <div
      class="TipCalculator-Info"
    >
      <h6>
        Tip
      </h6>
      <h6
        class="TipCalculator-Info-Amount"
      >
        $0.00
      </h6>
    </div>
    <div
      class="TipCalculator-ButtonGroup"
    >
      <button>
        12%
      </button>
      <button>
        15%
      </button>
      <button>
        18%
      </button>
      <input
        class="TipCalculator-CustomTip"
        name="custom-tip"
        value="0"
      />
    </div>
  </div>
</DocumentFragment>
`;
